numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -1114,7 +1114,13 @@ PyUFunc_DivisionTypeResolver(PyUFuncObject *ufunc,
             }
             out_dtypes[1] = out_dtypes[0];
             Py_INCREF(out_dtypes[1]);
+
+            if (strcmp(ufunc->name, "floor_divide") == 0) {
```

Comment:
Ideally you'd split the function into a floordiv resolver and a truediv resolver, and extract the common pieces to helper functions. I can't think of any issues this will cause right now though. Maybe just stick a todo here.

Reply:
added the todo